services/debugger/uploader.go

```diff
@@ -57,25 +61,35 @@ func New(url string, transformer Transformer) UploaderI {
 	eventBuffer := make([]interface{}, 0)
 	client := &http.Client{}
 
-	uploader := &Uploader{url: url, transformer: transformer, eventBatchChannel: eventBatchChannel, eventBuffer: eventBuffer, Client: client}
+	uploader := &Uploader{url: url, transformer: transformer, eventBatchChannel: eventBatchChannel, eventBuffer: eventBuffer, Client: client, bgWaitGroup: sync.WaitGroup{}}
 	uploader.Setup()
 	return uploader
 }
 
 func (uploader *Uploader) Start() {
+	ctx, cancel := context.WithCancel(context.Background())
+
 	rruntime.Go(func() {
 		uploader.handleEvents()
+		cancel()
 	})
+
+	uploader.bgWaitGroup.Add(1)
 	rruntime.Go(func() {
```

Comment:
Why are we using `rruntime.Go` instead of simply doing `go`?

Reply:
Afaik the plan is to remove this in the near future. Also, this discussion is outside the scope of this PR, and should be discussed independently, so I'll resolve this for now

Reply:
Sure, I was just asking why it's being done this way, not asking for changing it here.
Although I must say that given that the plan is to remove them you could consider dropping this one now :slightly_smiling_face: 

Up to you :+1: